### DISCUSSION TITLE
MQTT sends too many messages all the time

### DISCUSSION THREAD

**Discussion by ivanvach** [tagyoureit/nodejs-poolController#659] [App and O/S Config Questions]
How to make MQTT send out messages about the status only when status changes.
For me MQTT keeps sending a barrage of MQTT messages every 5 seconds about the status of ALL pool circuits although they are the same. No changes.
Here is my MQTT setup. As you can see "changesOnly" is set to true, but it doesn't help.

"mqtt": {
        "name": "MQTT",
        "type": "mqtt",
        "enabled": true,
        "fileName": "mqtt.json",
        "globals": {},
        "options": {
          "protocol": "mqtt://",
          "host": "192.168.1.222",
          "port": 1883,
          "username": "xxxx",
          "password": "xxxxxxxxxxx",
          "selfSignedCertificate": false,
          "rootTopic": "pool",
          "retain": false,
          "qos": 0,
          "changesOnly": true
        },

**Comment by tagyoureit** [comment]
Can you post a couple of example MQTT payloads that are being sent out?  I'd like to see exactly which they are.

**Comment by ivanvach** [comment]
Sure. These two came within 1 second interval. Please note that the last one has numerical value  as well.

pool/state/pumps/3/pump3/rpm    -->  {"rpm":null}
pool/state/pumps/3/pump3/rpm    -->  {"rpm":null}
pool/state/pumps/3/pump3/rpm    -->  {"rpm":0}

Unfortunately my MQTT client is not copy/paste friendly but the list is very long of repeating messages.

**Reply by tagyoureit** [reply]
Since we aren't picking up your 3rd pump, every time we get a message the app thinks there is a change.  Once we get the third pump sorted it this should only change when the data changes.
